department-of-veterans-affairs/vets-website
# VACMS-18796 Events search: remove duplicate analytics

## Are you removing, renaming or moving a folder in this PR?

## Summary
Remove duplicate analytics events when filtering for events. Previously, we logged events if the filters were applied incorrectly or the fields weren't filled out in addition to logging when the "Apply filter" button was clicked. Now we only have a single event that is fired when the filter button is used correctly to track the information that people are using to filter.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18796

## Testing done

1. Go to `/outreach-and-events/events`
2. Select "Custom date range" for `Filter by`
3. Don't fill out any fields and click "Apply filter"
4. Observe that no analytics events are fired and the component is in an error state (red line on the left side)
5. Fill out the fields incorrectly (e.g. end date that happens before the start date) and click "Apply filter"
6. Observe that no analytics events are fired and the component is in an error state ("Please select an end date that comes after the start date." message is visible)
7. Fill out the fields correctly and click "Apply filter"
8. Observe that the error state is cleared and you can see the below analytics event in the console

## Screenshots
<img width="628" alt="Screenshot 2024-08-07 at 6 11 31 PM" src="https://github.com/user-attachments/assets/ac22f3be-2ef3-4306-abad-1e13a589aa26">

<img width="531" alt="Screenshot 2024-08-07 at 6 12 04 PM" src="https://github.com/user-attachments/assets/052ba812-8678-4188-8865-d447e2479123">
